node/src/accountant/mod.rs

```diff
@@ -4221,80 +4206,89 @@ mod tests {
-            }
+    fn gap_tester(payment_thresholds: &PaymentThresholds) -> Option<u64> {
+        let cached = (0_u64..20).map(|to_add| {
+            ThresholdUtils::calculate_finite_debt_limit_by_age(&payment_thresholds, 1500 + to_add)
+                as u64
+        });
+        let mut counts_of_unique_elements: HashMap<u64, usize> = HashMap::new();
+        cached.for_each(|point_height| {
+            counts_of_unique_elements
+                .entry(point_height)
+                .and_modify(|q| *q += 1)
+                .or_insert(1);
+        });
+        let mut counts_of_groups_of_the_same_size: HashMap<usize, usize> = HashMap::new();
+        counts_of_unique_elements.values().for_each(|unique_count| {
+            counts_of_groups_of_the_same_size
+                .entry(*unique_count)
+                .and_modify(|q| *q += 1)
+                .or_insert(1);
+        });
+        let mut sortable = counts_of_groups_of_the_same_size
+            .drain()
+            .collect::<Vec<_>>();
+        sortable.sort_by_key(|(_key, count)| *count);
+        let (biggest_groups_size, occurrence) = sortable.last().expect("no values to analyze");
+        //checking if the sample of undistorted results (consist size groups) has enough weight compared to 20 tries from the beginning
+        if biggest_groups_size * occurrence >= 15 {
+            Some(*biggest_groups_size as u64)
+        } else {
+            panic!("couldn't provide a relevant amount of data for the analysis")
         }
+    }
 
+    fn test_height_granularity_with_advancing_time(
+        test_scope: &str,
+        payment_thresholds: &PaymentThresholds,
+        seconds_between_height_change: u64,
+    ) {
+        let finding_under_135_degree = gap_tester(&payment_thresholds);
+
+        if let Some(seconds_needed_for_smallest_detected_change_in_height) =
+            finding_under_135_degree
         {
-            let payment_thresholds = PaymentThresholds {
-                maturity_threshold_sec: 100,
-                payment_grace_period_sec: 0,
-                permanent_debt_allowed_gwei: 100,
-                debt_threshold_gwei: 10_000,
-                threshold_interval_sec: 10_000,
-                unban_below_gwei: 100,
-            };
+            assert_eq!(seconds_needed_for_smallest_detected_change_in_height, seconds_between_height_change,
+                       "while testing {} we expected that these thresholds: {:?} will require only {} s until we see the height change but computed {} s instead",
+                       test_scope, payment_thresholds, seconds_between_height_change, seconds_needed_for_smallest_detected_change_in_height)
+        } else {
+            panic!(
+                "while testing {}, we waited for some finding but got none",
+                test_scope
+            )
+        }
+    }
 
-            let finding_under_135_degree = gap_tester(&payment_thresholds);
+    #[test]
```

Comment:
Maybe also a test with a shallower slope proving that in such a case you needed more seconds to get a one-wei change in limit.